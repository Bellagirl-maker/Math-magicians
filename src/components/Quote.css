.quoteDiv {
  width: 80%;
  font-weight: 700;
  text-align: center;
  padding: 2rem;
  margin: 5rem auto 0;
}
