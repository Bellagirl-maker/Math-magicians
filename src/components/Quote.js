import React, { useState, useEffect } from 'react';
import './Quote.css';

const apiKey = 'PvhIKqsrHwPYCrtU+wSBpQ==w8E2CqRsF9rkbWmK';
const apiBaseURL = 'https://api.api-ninjas.com/v1/quotes?category=';
const apiCategory = 'computers';

export default function Quote() {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchApi = async () => {
      setLoading(true);
      try {
        const res = await fetch(`${apiBaseURL}${apiCategory}`, {
          method: 'GET',
          headers: { 'X-Api-Key': `${apiKey}` },
          contentType: 'application/json',
        });
        const json = await res.json();
        setQuote(json[0]);
        setLoading(false);
      } catch (err) {
        setError(true);
      }
      setLoading(false);
    };

    fetchApi();
  }, []);

  if (error) {
    return (
      <div className="quoteDiv">
        <p className="quote">Something went wrong!</p>
      </div>
    );
  }

  if (loading) {
    return (
      <div className="quoteDiv">
        <p className="quote">Loading...</p>
      </div>
    );
  }

  return (
    <div className="quoteDiv">
      <p className="quote">
        &quot;
        {quote.quote}
        &quot;
      </p>
      <cite className="author">{quote.author}</cite>
    </div>
  );
}
