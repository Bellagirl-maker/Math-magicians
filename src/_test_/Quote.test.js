import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';
import '@testing-library/jest-dom/extend-expect';

describe('Quote Component', () => {
  it('renders the Quote component', async () => {
    render(<Quote />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });
});

describe('Quote snapshot', () => {
  test('Quote snapshot should match dom', () => {
    const tree = renderer.create(<Quote />).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
