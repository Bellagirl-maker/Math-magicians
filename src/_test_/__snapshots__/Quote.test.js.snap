// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote snapshot Quote snapshot should match dom 1`] = `[Function]`;
